### ISSUE TITLE
PLW1641/`eq-without-hash` should not trigger when explicitly setting `__hash__ = None`

### ISSUE THREAD

**Issue by bluetech** [astral-sh/ruff#6701]
Given

```py
class MyClass:
    def __eq__(self, other):
        return True

    __hash__ = None
```

Running `ruff --isolated --select PLW1641` with Ruff 0.0.285:

```
x.py:1:7: PLW1641 Object does not implement `__hash__` method
```

While the purpose of `PLW1641` is exactly to check for when `__eq__` is defined and `__hash__` is `None` (the default when `__eq__` is defined but `__hash__` is not), I think that if the code explicitly sets `__hash__ = None` then the lint should not trigger since the programmer explicitly asks for it.
